cpu/esp32/Kconfig

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) 2020 HAW Hamburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+config CPU_CORE_XTENSA_LX6
+    bool
+    select CPU_ARCH_XTENSA
+
+config CPU_FAM_ESP32
+    bool
+    select CPU_CORE_XTENSA_LX6
+    select CPU_COMMON_ESP
+    select HAS_ARCH_ESP32
+    select HAS_CPU_ESP32
+    select HAS_ESP_WIFI_ENTERPRISE
+    select HAS_PERIPH_ADC_CTRL
+    select HAS_PERIPH_RTC
+    select HAS_PERIPH_RTT
+
+## CPU Models
+config CPU_MODEL_ESP32_WROOM_32
+    bool
+    select CPU_FAM_ESP32
+
+config CPU_MODEL_ESP32_WROVER
+    bool
+    select CPU_FAM_ESP32
+
+config CPU_MODEL_ESP32_D0WDQ6
+    bool
+    select CPU_FAM_ESP32
+
+## Definition of specific features
+config HAS_ARCH_ESP32
+    bool
+    help
+        Indicates that the current architecture is ESP32.
+
+config HAS_CPU_ESP32
+    bool
+    help
+        Indicates that the current CPU is 'esp32'.
+
+config HAS_ESP_RTC_TIMER_32K
+    bool
+    help
+        Indicates that an external 32.768 kHz crystal is connected to the ESP32
+        in the board.
+
+config HAS_ESP_SPI_RAM
+    bool
+    help
+        Indicates that an external RAM is connected via the FSPI interface in
+        the board.
+
+config HAS_ESP_WIFI_ENTERPRISE
+    bool
+    help
+        Indicates that the platform supports WPA2 enterprise mode for the WiFi
+        interface.
+
+config HAS_PERIPH_ADC_CTRL
+    bool
+    help
+        Indicates that an ESP32 ADC controller peripheral is present.
+
```

Comment:
ADC controller can't be configured by the user. It should be exposed in Kconfig.
```suggestion
```

Reply:
But this is a feature provided

Reply:
As you can see the symbol has no prompt, meaning that the user can't change the value (it's an invisible symbol). It's here in order to indicate that the feature is provided, like indicated in `Makefile.features`.

Reply:
I don't remember exactly, but providing it as a feature was a trick to compile `cpu/esp32/periph/adc_ctrl.c` when it is needed by other peripherals
https://github.com/RIOT-OS/RIOT/blob/ef4835a0927bff694753435b8fbdc425c0125b6c/cpu/esp32/Makefile.dep#L55-L57 As I remember, the C files in `cpu/*/periph` are compiled based on used features.

Reply:
> As I remember, the C files in cpu/*/periph are compiled based on used features.

Yes. That's why this is included in the Kconfig. All features provided by `Makefile.features` should match these.

Reply:
Hm, but `periph_adc_ctrl` is only used internally and shouldn't be exposed to the user. Isn't it possible to drop this feature from Kconfig?

Reply:
As I mentioned, the user does not see this symbol and can't change the value of it directly (it has no prompt). It is there just to indicate that the feature is provided, just as `FEATURES_PROVIDED += periph_adc_ctrl` is there.

Reply:
> As I mentioned, the user does not see this symbol 

It is shown in the feature list
```
-*- <HAS_PERIPH_ADC_CTRL>
```

Reply:
In order to be able to see that, one needs to enable the `show-all` mode in menuconfig. That means that one will see symbols with and without prompt. You must have noticed also that one can't change the value of it. There is no way to make it not appear at all when using this mode.